setup-ssl.sh

```diff
@@ -11,11 +11,22 @@ fi
 
 # /etc/ssl/private can't be accessed from within container for some reason
 # (@andrewgodwin says it's something AUFS related)  - taken from https://github.com/orchardup/docker-postgresql/blob/master/Dockerfile
-cp -r /etc/ssl /tmp/ssl-copy/
+
+
+
+if [[ ! -d /configs ]]; then
+  echo "Creating SSL config directory at /configs"
+  mkdir -p /configs
+fi
+if [ ! -f /configs/self-signed-ssl.key -o ! -f /configs/self-signed-ssl.pem ] ; then
+    openssl req -x509 -newkey rsa:4096 -keyout /configs/self-signed-ssl.key -out /configs/self-signed-ssl.pem -days 3650 -nodes -sha256 -subj '/CN=gis'
+
+fi
+cp /configs/self-signed-ssl.pem /etc/ssl/certs/ssl-cert-snakeoil.pem
+cp /configs/self-signed-ssl.key /etc/ssl/private/ssl-cert-snakeoil.key
```

Comment:
Hard coded destination but configurable through env vars?

Reply:
My thinking was if we used env variables like `SSL_CERT_FILE=/tmp/certs/ssl_cert.pem`  the permissions would not be applied and also conversely if I mount a file to `/etc/ssl/certs/ssl_cert.pem` then it won't work because of the permissions but if we mount it to a different fixed destination then we could have logic to apply for the permissions.